### PR TITLE
vendor-install.sh: Fix `shellcheck` style errors

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#11086

### CODE REVIEW

File: Library/Homebrew/cmd/vendor-install.sh [hunk 1]
```diff
@@ -4,42 +4,49 @@
 #:  Install Homebrew's portable Ruby.
 
 # Don't need shellcheck to follow this `source`.
-# shellcheck disable=SC1090
-source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
+# set from user environment
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/vendor-install.sh:7]:
```suggestion
# HOMEBREW_LIBRARY set by/brew
```

File: Library/Homebrew/cmd/vendor-install.sh [hunk 2]
```diff
@@ -4,42 +4,49 @@
...
 
-VENDOR_DIR="$HOMEBREW_LIBRARY/Homebrew/vendor"
+VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 #
 # Dynamic variables can't be detected by shellcheck
-# shellcheck disable=SC2034
-if [[ -n "$HOMEBREW_MACOS" ]]
+# set from user environment
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/vendor-install.sh:16]: This (and most others) isn't set from user environment; can you check the definitions of these?

**Comment by MikeMcQuaid** [comment]
Nice work so far @GauthamGoli!

**Comment by MikeMcQuaid** [comment]
@hyuraku would you be able to cherry-pick the commit from here and finish off this PR in your own? Thanks ❤️

**Comment by hyuraku** [comment]
> @hyuraku would you be able to cherry-pick the commit from here and finish off this PR in your own? Thanks ❤️

@MikeMcQuaid 
OK!, I will finish the PR with cherry-pick

**Comment by Bo98** [comment]
Handled in #11238.

**Pull request closed without merging**